plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/ModelExecutorThread.java

```diff
@@ -76,24 +79,20 @@ private void newMailboxEntry(MailboxEntry entry) {
 	 * becomes empty.
 	 */
 	void shutdown() {
-		try {
-			class ShutdownAction implements MailboxEntry {
-				@Override
-				public void execute() {
-					if (mailbox.isEmpty()) {
-						ModelExecutor.shutdownNow();
-					} else {
-						try {
-							mailbox.put(new ShutdownAction());
-						} catch (InterruptedException e) {
-						}
-					}
-				}
-			}
+		shutdownWhenEmpty = true;

```

Comment:
I don't think there could be synchronization problems since the method only sets the flag to true. There is no method to set this flag to anything except true.